src/__tests__/labs/ViewLab.test.tsx

```diff
@@ -66,160 +66,149 @@ describe('View Lab', () => {
+          }),
+        ).toHaveValue(expectedLab.result)
+      })
     })
 
     it('should display the past notes', async () => {
       const expectedNotes = 'expected notes'
       const expectedLab = { ...mockLab, notes: [expectedNotes] } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
+      const { container } = setup(expectedLab, [Permissions.ViewLab])
 
-      const notes = wrapper.findWhere((w) => w.prop('data-test') === 'note')
-      const pastNotesIndex = notes.reduce(
-        (result: number, item: ReactWrapper, index: number) =>
-          item.text().trim() === expectedNotes ? index : result,
-        -1,
-      )
-
-      expect(pastNotesIndex).not.toBe(-1)
-      expect(notes).toHaveLength(1)
+      await waitFor(() => {
+        expect(screen.getByText(expectedNotes)).toBeInTheDocument()
+        expect(container.querySelector('.callout')).toBeInTheDocument()
+      })
     })
 
     it('should not display past notes if there is not', async () => {
       const expectedLab = { ...mockLab, notes: undefined } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
-
-      const notes = wrapper.findWhere((w) => w.prop('data-test') === 'note')
+      const { container } = setup(expectedLab, [Permissions.ViewLab])
 
-      expect(notes).toHaveLength(0)
+      await waitFor(() => {
+        // TODO: make sure it's not loading
+        expect(container.querySelector('.callout')).not.toBeInTheDocument()
+      })
     })
 
     it('should display the notes text field empty', async () => {
       const expectedNotes = 'expected notes'
       const expectedLab = { ...mockLab, notes: [expectedNotes] } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab])
+      setup(expectedLab, [Permissions.ViewLab])
 
-      const notesTextField = wrapper.find(TextFieldWithLabelFormGroup).at(1)
-
-      expect(notesTextField).toBeDefined()
-      expect(notesTextField.prop('value')).toEqual('')
+      await waitFor(() => {
+        expect(screen.getByLabelText('labs.lab.notes')).toHaveValue('')
+      })
     })
 
     it('should display errors', async () => {
       const expectedLab = { ...mockLab, status: 'requested' } as Lab
-      const { wrapper } = await setup(expectedLab, [Permissions.ViewLab, Permissions.CompleteLab])
+      setup(expectedLab, [Permissions.ViewLab, Permissions.CompleteLab])
 
       const expectedError = { message: 'some message', result: 'some result feedback' } as LabError
       expectOneConsoleError(expectedError)
       jest.spyOn(validateUtil, 'validateLabComplete').mockReturnValue(expectedError)
 
-      const completeButton = wrapper.find(Button).at(1)
-      await act(async () => {
-        const onClick = completeButton.prop('onClick') as any
-        await onClick()
+      await waitFor(() => {
+        userEvent.click(
+          screen.getByRole('button', {
+            name: /labs\.requests\.complete/i,
```

Comment:
Just curios, few places I see . & few other \.

Escaping is needed?
`screen.getByLabelText(/labs.lab.result/i)`

Reply:
I got this query from https://testing-playground.com/